### PR TITLE
FixingTheDocs

### LINKED ISSUE #147: Documentation build is broken on RTD
## Description

RTD complains of excessive memory consumption. Mock imports along with a custom environment file for the doc build could help.

### PR DESCRIPTION
**Pull request by Zeitsperre**
## Overview

This PR fixes #147 

Changes:

* Added some mocked imports and configurations to sphinx conf.py
* First pass at a .travis.yml update
* Updated the ReadTheDocs.yml to employ version 2 of API

## Additional Information

Links to other issues or sources.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -18,7 +18,11 @@
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-sys.path.insert(0, os.path.abspath('../../'))
+sys.path.insert(0, os.path.abspath('../..'))
+
+# Set flag to not fail doc build.
+if 'DO_NOT_CHECK_EXECUTABLE_EXISTENCE' not in os.environ:
```
Comment (tlvu) [inline docs/source/conf.py]: I had added this 'DO_NOT_CHECK_EXECUTABLE_EXISTENCE' for Raven doc only.  This is birdy doc, I do not think you need this here.

**Comment by Zeitsperre** [comment]
@tlvu I can't seem to see any projects related to bird-house in RtD. It could be that I need to be added to the list of maintainers on that site.

@huard or @cehbrecht can you add me to the bird-house organization for RtD?

**Comment by huard** [comment]
@Zeitsperre done.

**Comment by Zeitsperre** [comment]
Docs are now building and tests are passing on Travis. The setup for Travis isn't perfect but it works for now.

**Comment by huard** [comment]
@Zeitsperre In the future please wait for Carsten's approval before merging.

**Comment by Zeitsperre** [comment]
Alright. Will do.

**Comment by cehbrecht** [comment]
@Zeitsperre thanks for the fix :)

**Pull request merged**